### DISCUSSION TITLE
Swift SDK

### DISCUSSION THREAD

**Discussion by gsabran** [modelcontextprotocol/.github#100] [Show and tell]
### Pre-submission Checklist

- [X] This post follows the MCP community guidelines

### What would you like to share?

I started writing a [Swift SDK](https://github.com/gsabran/mcp-swift-sdk). There's only a client for now - which should match how MCP would most often be used in Swift. I'm planning to refine it, add a server SDK, and a demo host app.

Any contributions or feedback is welcome!

I saw that a Kotlin SDK has been added to the MCP repo, while other Rust/Go SDKs are being developed by the community. Any guidance from MCP folks on what will remain within the community vs up-leveled to this repo?

### Relevant Links

Github repo: https://github.com/gsabran/mcp-swift-sdk

**Comment by yujinqiu** [comment]
@gsabran Great job <3

**Comment by gsabran** [comment]
Added a server SDK as well.

**Comment by jspahrsummers** [comment]
Awesome to see, thanks for sharing!

> I saw that a Kotlin SDK has been added to the MCP repo, while other Rust/Go SDKs are being developed by the community. Any guidance from MCP folks on what will remain within the community vs up-leveled to this repo?

At the moment, we're focusing on implementations that have significant organizational support (not necessarily Anthropic's), as this tends to guarantee a higher level of maintenance and community engagement. This isn't super formalized right now, but hopefully it makes sense.

**Reply by gsabran** [reply]
Could you shed some light into what went in the decision to prefer this other package, that doesn't seem to have more organizational support?

https://github.com/modelcontextprotocol/swift-sdk/pull/28
